### ISSUE TITLE
Seeking Feedback and Timeline for B&A/KV Support for Azure

### ISSUE THREAD

**Issue by Brandr0id** [WICG/protected-auction-services-discussion#90]
Currently KMS, B&A, and KV services support GCP and AWS without support for the second largest cloud provider (Azure). Implementations for AWS and GCP predated any publicly-defined process/mechanism and were approved ~7mo ago in April 2024 (https://github.com/WICG/protected-auction-services-discussion/issues/63, https://github.com/WICG/protected-auction-services-discussion/issues/64). Support for Azure has been requested numerous times, via multiple channels as has been discussed in (https://github.com/WICG/protected-auction-services-discussion/issues/31) with a formal request made in this thread **~6mo** ago with a quick ack on reviewing but no formal approval or updates.

In parallel the Chrome browser has carried out Origin Trials, and is looking to ship the B&A APIs as we speak ([Blink-Dev Intent to Ship Bidding & Auction APIs](https://groups.google.com/a/chromium.org/g/blink-dev/c/CyXJz3xOEcc/m/1snkv2KhAwAJ?utm_medium=email&utm_source=footer)). Given Chrome efforts to ship the corresponding B&A APIs we are deeply concerned that it will leave no chance for ad tech entities that depend on Azure to evaluate these solutions in Azure or provide meaningful feedback.

Our Azure R&D and security experts have had initial PRs open to provide either direct contributions, or fully functionally example code/implementations to the team (however you wish to use this) for **>11mo with no comments, feedback, or other action**: PR: https://github.com/privacysandbox/data-plane-shared-libraries/pull/1. For convenience a new PR has been updated with the latest/complete Azure CPIO/dataplane support rebased to the HEAD of the dataplane branch. We are happy to promptly address feedback, make direct contributions (split the PR up as needed), or provide consultation with the specifics of how to adapt the existing code for operation in Azure.

Given this background, @chatterjee-priyanka , @renanfel , can you and the team please help provide feedback on timelines for supporting Azure, how it integrates with any efforts we've heard about relating KMS/coordinator re-architecture (https://github.com/WICG/protected-auction-services-discussion/issues/69), and what concerns or impediments there may be to accepting direct contributions to start unblocking things in this space by contributors already covered by a Google CLA?

**Comment by bmayd** [comment]
It does seem that the approval of Azure is taking an inordinate amount of time with very little visibility into why that’s the case, particularly when considered in light of the fact that AWS and GCP were approved within days of the posting of [Public Cloud TEE Requirements](https://github.com/privacysandbox/protected-auction-services-docs/blob/main/public_cloud_tees.md#public-cloud-tee-requirements) in April this year. Presumably those implementations were largely completed before the requirements were made public and during the period when the Azure team was regularly requesting direction from the Sandbox team on how to participate in the process. It's been clear to me from the beginning of the effort this time last year, and I had assumed also to everyone engaged in the discussions, that TEE services support from Azure is critical to the ecosystem and that the Azure team was both fully committed to providing that support and entirely willing work with the Sandbox team, and within its process, to meet Sandbox requirements. It is hard to understand why progress has been so slow.

In case it isn’t clear to folks why Azure support is critical: the only cloud providers currently approved by the Sandbox team are AWS and GCP. It is very common for adtech service providers to be told by clients, in no uncertain terms, that they will not do business with partners that support Amazon or use AWS, as many advertisers consider Amazon to be a competitor. If the only available TEE options are AWS and GCP, most of adtech will effectively have no choice but to use GCP and while it is generally unhealthy for an ecosystem to have a critical dependency on a single supplier, that is particularly true in the case of adtech which is already overly dependent on Google.

In the [Public Cloud TEE Requirements: Review Process](https://github.com/privacysandbox/protected-auction-services-docs/blob/main/public_cloud_tees.md#review-process) section it says:

> We aim for the review process and discussion to be transparent.

and

> We will provide appropriate notice when we decide which CSPs to support, and the expected timeline (including any changes to the timeline as appropriate).

Efforts are falling short in both cases and the lack of transparency, clear guidance and timelines from the Sandbox team for Azure support are increasingly threatening to put adtech businesses using Azure, and the the advertising clients that depend on them, at a competitive disadvantage. Can we get better feedback from the Sandbox team regarding support for Azure?

**Comment by renanfel** [comment]
We have posted an update on the request for supporting Azure on the original [github post](https://github.com/WICG/protected-auction-services-discussion/issues/31#issuecomment-2555792479). As noted there, we plan to support B&A on Azure in-line with the [Cross-Cloud Coordinator explainer](https://github.com/privacysandbox/protected-auction-services-docs/blob/main/cross_cloud_coordinators.md), and implement B&A using Azure TEEs in consultation with Microsoft. Subject to additional diligence and coordination with Microsoft, we expect to reach the Alpha milestone for B&A on Azure in Q3 2025, and the Beta milestone 6 months after ad techs test the Alpha version and provide feedback.
